song-server/src/main/java/org/icgc/dcc/song/server/service/AnalysisService.java

```diff
@@ -75,15 +96,19 @@ public String create(String studyId, Analysis a) {
      repository.createVariantCall(experiment);
    } else {
      // shouldn't be possible if we validated our JSON first...
-     return "Analysis failed: Unknown Analysis Type";
+     throw new IllegalArgumentException("Invalid analysis type");
    }
+   return id;
+  }
 
-  return id;
+  public String updateAnalysis(String studyId, Analysis a) {
+    assert false; // not coded yet
```

Comment:
create a ticket for this, then add a comment
`//TODO:  [DCC-1234]  implement update analysis`